src/httprpc.cpp

```diff
@@ -223,18 +236,49 @@ static bool HTTPReq_JSONRPC(const std::any& context, HTTPRequest* req)
                     }
                 }
             }
-            strReply = JSONRPCExecBatch(jreq, valRequest.get_array());
+
+            // Execute each request
+            reply = UniValue::VARR;
+            for (size_t i{0}; i < valRequest.size(); ++i) {
+                // Batches never throw HTTP errors, they are always just included
+                // in "HTTP OK" responses. Notifications never get any response.
+                UniValue response;
+                try {
+                    jreq.parse(valRequest[i]);
+                    response = JSONRPCExec(jreq, /*catch_errors=*/true);
+                } catch (UniValue& e) {
+                    response = JSONRPCReplyObj(NullUniValue, std::move(e), jreq.id, jreq.m_json_version);
+                } catch (const std::exception& e) {
+                    response = JSONRPCReplyObj(NullUniValue, JSONRPCError(RPC_PARSE_ERROR, e.what()), jreq.id, jreq.m_json_version);
+                }
```

Comment:
(Came across this block again while working on something else. It struck me that the exception handling looked redundant as we catch the same exact exception types inside `JSONRPCExec(jreq, /*catch_errors=*/true`). *But* the `jreq.parse()` call will throw on missing methods etc.

One could move out `response = JSONRPCExec(jreq, /*catch_errors=*/true);` to after the block, but then one would need to guard against using a `jreq` that failed to parse. So the current version of the block is probably preferable both in readability and efficiency).